pkg/storage/unified/backend/errors.go

```diff
@@ -0,0 +1,21 @@
+package backend
```

Comment:
I'm imagining that the backend interface and similar might also move to this package, depending on whether we want the implementations to assert they implement it.

I couldn't immediately see a better package to put what I put into this, though perhaps we have one?